libraries/AP_BattMonitor/AP_BattMonitor_SMBus_NeoDesign.cpp

```diff
@@ -0,0 +1,102 @@
+#include <AP_HAL/AP_HAL.h>
+#include <AP_Common/AP_Common.h>
+#include <AP_Math/AP_Math.h>
+#include <AP_Notify/AP_Notify.h>
+#include "AP_BattMonitor.h"
+#include "AP_BattMonitor_SMBus_NeoDesign.h"
+#include <utility>
+
+#define BATTMONITOR_ND_CELL_COUNT                0x5C    // cell-count register
+#define BATTMONITOR_ND_CELL_START                0x30    // first cell register
+#define BATTMONITOR_ND_CURRENT                   0x0a    // current register
+#define BATTMONITOR_ND_TEMPERATURE               0x08    // temperature register
+
+// Constructor
+AP_BattMonitor_SMBus_NeoDesign::AP_BattMonitor_SMBus_NeoDesign(AP_BattMonitor &mon,
+                                                   AP_BattMonitor::BattMonitor_State &mon_state,
+                                                   AP_BattMonitor_Params &params,
+                                                   AP_HAL::OwnPtr<AP_HAL::I2CDevice> dev)
+    : AP_BattMonitor_SMBus(mon, mon_state, params, std::move(dev))
+    , _cell_count(0)
+{
+    _pec_supported = false;
+}
+
+void AP_BattMonitor_SMBus_NeoDesign::timer()
+{
+    uint8_t buff[8];
+    uint32_t tnow = AP_HAL::micros();
+
+    // Get the cell count once. Not likely to change in flight, LOL
+    if (_cell_count == 0) {
+        if (read_block_bare(BATTMONITOR_ND_CELL_COUNT, buff, 2)) {
+            _cell_count = buff[1] << 8 | buff[0];
```

Comment:
Sanity check this to not be > 10 cells, as the data sheet only gives us registers for 10 cells, and obviously values like 512 will lead to crazy stuff happening later when we try and read the cells (as well as cause us to spend way to much time trying to read them)